content/posts/authz-feegrant-blog.md

```diff
@@ -0,0 +1,199 @@
+
+IdP address: `hid1qemjjsrsjwyuucv0zr370kj8k9dakvp9907zzj`
+
+1. User's account is created
+
+```
+$ hid-noded keys add user
+
+- name: user
+  type: local
+  address: hid1enragujgzhex39tt34ljtqmwgljn0kr685f3nk
+  pubkey: '{"@type":"/cosmos.crypto.secp256k1.PubKey","key":"A4+zpdVic3RpnJvATZBuhFsH9lPGy/q0+uGRqSici3yN"}'
+  mnemonic: ""
+```
+
+2. IdP authorizing User to perform DID Creation Transaction. We have to specify the TypeURL of the transaction being authorized.
+
+```
+hid-noded tx authz grant hid1enragujgzhex39tt34ljtqmwgljn0kr685f3nk generic --from hid1qemjjsrsjwyuucv0zr370kj8k9dakvp9907zzj --msg-type "/hypersignprotocol.hidnode.ssi.MsgCreateDID"
+```
+
+We can list the authorizations being granted to User by IdP:
+
+```
+$ hid-noded q authz grants hid1qemjjsrsjwyuucv0zr370kj8k9dakvp9907zzj hid1enragujgzhex39tt34ljtqmwgljn0kr685f3nk
+
+grants:
+- authorization:
+    '@type': /cosmos.authz.v1beta1.GenericAuthorization
+    msg: /hypersignprotocol.hidnode.ssi.MsgCreateDID
+  expiration: "2023-04-14T00:35:04Z"
+pagination:
+  next_key: null
+  total: "0"
+```
+
+3. IdP Grants User a spend limit of `1000uhid` for Transaction Fee.
+
+```
+hid-noded tx feegrant grant hid1qemjjsrsjwyuucv0zr370kj8k9dakvp9907zzj hid1enragujgzhex39tt34ljtqmwgljn0kr685f3nk --spend-limit 1000uhid
+```
+
+We can check the allowance that was given to User:
+
+```
+$ hid-noded q feegrant grants hid1enragujgzhex39tt34ljtqmwgljn0kr685f3nk
+
+allowances:
+- allowance:
+    '@type': /cosmos.feegrant.v1beta1.BasicAllowance
+    expiration: null
+    spend_limit:
+    - amount: "1000"
+      denom: uhid
+  grantee: hid1enragujgzhex39tt34ljtqmwgljn0kr685f3nk
+  granter: hid1qemjjsrsjwyuucv0zr370kj8k9dakvp9907zzj
+pagination:
+  next_key: null
+  total: "0"
+```
+
+4. User generates DID Document from the SDK, signs it with their pivate keys and sends it to Idp. Since, we are focused only on CLI, we will be executing the similar SDK process with its help.
+
+```
+hid-noded tx ssi create-did '{
+"context": [
+"https://www.w3.org/ns/did/v1",
+"https://w3id.org/security/v1",
+"https://schema.org"
+],
+"id": "did:hs:0f49341a-20ef-43d1-bc93-de30993e6c51",
+"controller": ["did:hs:0f49341a-20ef-43d1-bc93-de30993e6c51"],
+"verificationMethod": [
+{
+"id": "did:hs:0f49341a-20ef-43d1-bc93-de30993e6c51#zEYJrMxWigf9boyeJMTRN4Ern8DJMoCXaLK77pzQmxVjf",
+"type": "Ed25519VerificationKey2020",
+"controller": "did:hs:0f49341a-20ef-43d1-bc93-de30993e6c51",
+"publicKeyMultibase": "zEYJrMxWigf9boyeJMTRN4Ern8DJMoCXaLK77pzQmxVjf"
+}
+],
+"authentication": [
+"did:hs:0f49341a-20ef-43d1-bc93-de30993e6c51#zEYJrMxWigf9boyeJMTRN4Ern8DJMoCXaLK77pzQmxVjf"
+]
+}' did:hs:0f49341a-20ef-43d1-bc93-de30993e6c51#zEYJrMxWigf9boyeJMTRN4Ern8DJMoCXaLK77pzQmxVjf --ver-key oVtY1xceDZQjkfwlbCEC2vgeADcxpgd27vtYasBhcM/JLR6PnPoD9jvjSJrMsMJwS7faPy5OlFCdj/kgLVZMEg== --from hid1enragujgzhex39tt34ljtqmwgljn0kr685f3nk --generate-only > tx.json
+```
+
+5. User submits the transaction with IdP acting as the fee Payer.
```

Comment:
Technically, its the User that signs the transaction and submits to blockchain. Only the role of paying fees is taken care by IdP. There needs to a change in Sequence Diagram, where this particular should be from User to Blockchain, instead of IdP to Blockchain. Will make the necessary ammends